File: babyshark.hs
module Main where

main :: IO ()
main =
  putStr $
  j
    "\n"
    [ j "" (replicate 3 $ s ++ ", doo doo doo doo doo doo\n") ++ s ++ "!"
    | s <-
        [ "Baby shark"
        , "Mommy shark"
        , "Daddy shark"
        , "Grandma shark"
        , "Grandpa shark"
        , "Let's go hunt"
        , "Run away"
        , "Safe at last"
        , "It's the end"
        ]
    ]

j :: [a] -> [[a]] -> [a]
j _ [i] = i
j d (i:ix) = i ++ d ++ j d ix

File: babyshark.java
class Babyshark {
    static String[] verses = {
        "Baby shark",
        "Mommy shark",
        "Daddy shark",
        "Grandma shark",
        "Grandpa shark",
        "Let's go hunt",
        "Run away",
        "Safe at last",
        "It's the end"
    };
    static String doodoo = ", doo doo doo doo doo doo";
    
    public static void main(String[] args) {
        for (String verse : verses) {
            System.out.println(verse + doodoo);
            System.out.println(verse + doodoo);
            System.out.println(verse + doodoo);
            System.out.println(verse + "!");
        }
    }
}
